cpp/src/arrow/testing/random.cc

```diff
@@ -867,6 +869,7 @@ std::shared_ptr<Array> RandomArrayGenerator::ArrayOf(const Field& field, int64_t
       // This isn't as flexible as it could be, but the array-of-structs layout of this
       // type means it's not a (useful) composition of other generators
       GENERATE_INTEGRAL_CASE_VIEW(Int64Type, DayTimeIntervalType);
+      GENERATE_INTEGRAL_CASE_VIEW(Int64Type, MonthDayNanoIntervalType);
```

Comment:
Hmm... it seems you should view a `fixed_size_binary(16)` instead.
Also, can you add some tests in `random_test.cc`?

Reply:
should be done.

Reply:
sorry still need to add a test.